library/std/src/sys/uefi/args.rs

```diff
@@ -0,0 +1,152 @@
+use r_efi::protocols::loaded_image;
+
+use crate::env::current_exe;
+use crate::ffi::OsString;
+use crate::fmt;
+use crate::iter::Iterator;
+use crate::mem::size_of;
+use crate::sys::uefi::helpers;
+use crate::vec;
+
+pub struct Args {
+    parsed_args_list: vec::IntoIter<OsString>,
+}
+
+pub fn args() -> Args {
+    // SAFETY: Each loaded image has an image handle that supports EFI_LOADED_IMAGE_PROTOCOL
+    let protocol =
+        helpers::image_handle_protocol::<loaded_image::Protocol>(loaded_image::PROTOCOL_GUID)
+            .unwrap();
+
+    let lp_size = unsafe { (*protocol.as_ptr()).load_options_size } as usize;
+
+    // Break if we are sure that it cannot be UTF-16
+    if lp_size < size_of::<u16>() || lp_size % size_of::<u16>() != 0 {
+        return Args {
+            parsed_args_list: Vec::from([current_exe().map(Into::into).unwrap_or_default()])
+                .into_iter(),
+        };
+    }
+
+    let lp_cmd_line = unsafe {
+        let lp_size = lp_size / size_of::<u16>();
+        let temp = (*protocol.as_ptr()).load_options as *const u16;
```

Comment:
Does something guarantee that temp is sufficiently aligned for u16 here?

Reply:
No, it is actually a `void *`.  Now the spec actually does seem to guarantee an alignment of `u8`, since it points to a variable size array as the last member of a c struct `_EFI_LOAD_OPTION`, so would that be enough along with the check for size being a multiple of 2?

Reply:
I would recommend explicitly checking this pointer's alignment (e.g., [`temp.is_aligned()`](https://doc.rust-lang.org/nightly/std/primitive.pointer.html#method.is_aligned)), or manually implementing that function.

Alignment of u8 isn't meaningful - everything is at least byte aligned.